API.md

```diff
@@ -0,0 +1,213 @@
+<!-- EDIT AT https://raw.githubusercontent.com/TheThingsNetwork/node-app-lib/master/API.md -->
+
+# API Reference
+
+Require the TTN Client module:
+
+```java
+import org.thethingsnetwork.java.app.lib.Client;
+import org.thethingsnetwork.java.app.lib.events.*;
+```
+
+## Class: Client
+
+Creates an instance of the client:
+
+```java
+Client client = new Client(region, appId, accessKey [, connOpts]);
+```
+
+* `region [String]`: The region (e.g. `eu`) or full hostname (e.g. `eu.thethings.network`) of the handler to connect to.
+* `appId [String]`: The ID of the application to connect to (e.g. `hello-world`).
+* `appAccessKey [String]`: An access key for the application, formatted as base64 (e.g. `'2Z+MU0T5xZCaqsD0bPqOhzA6iygGFoi4FAgMFgBfXSo='`).
+* `connOpts [MqttConnectOptions]`: Some custom configuration of the MQTT connection. This parameter is optional.
```

Comment:
For this part, there are several ways for accepting self-signed certificates, I'll just add a link to a complete doc of the different ways